### PR TITLE
keep coordinates encoding even when coordinates are decoded

### LINKED ISSUE #610: don't throw away variable specific coordinates information
Currently, we decode the `coordinates` attribute, when present, but it doesn't end up in the DataArray's `encoding` attribute (https://github.com/xray/xray/blob/master/xray/conventions.py#L822-L832).  This should be changed so the user can reference the `coordinates` attribute after decoding.

xref: #605

### PR DESCRIPTION
**Pull request by jhamman**
fixes #610

### CODE REVIEW

File: xray/conventions.py [hunk 1]
```diff
@@ -826,10 +826,10 @@ def stackable(dim):
         if decode_coords:
             var_attrs = new_vars[k].attrs
             if 'coordinates' in var_attrs:
-                var_coord_names = var_attrs['coordinates'].split()
+                coords = pop_to(var_attrs, new_vars[k].encoding, 'coordinates')
+                var_coord_names = coords.split()
                 if all(k in variables for k in var_coord_names):
                     coord_names.update(var_coord_names)
-                    del var_attrs['coordinates']
```
Comment (shoyer) [inline xray/conventions.py]: So, one cute thing about the existing code is that it doesn't delete the `coordinates` attribute unless all the coordinates are valid. That's what the `assertDatasetIdentical` test is doing that you changed to `assertDatasetEqual`.

Comment (jhamman) [inline xray/conventions.py]: I see.  So you're the approach you're recommending is to only put the coords in `.encoding['coordinates']` if the coordinates are valid.  That is easy enough.

Comment (shoyer) [inline xray/conventions.py]: Yes, exactly. WRF, for example, produces netCDF files with a `coordinates` attribute that refers to variables that don't exist in the file.

Comment (jhamman) [inline xray/conventions.py]: Well we could all wish WRF didn't do that but...

**Comment by jhamman** [comment]
I fixed this up. It should be good to go now.

**Pull request merged**